app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,50 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show new create]
```

Comment:
Are you sure that you don't need to check that a user is authenticated before creating a post? Because as an unauthenticated user I tried to access the path 'http://localhost:3000/posts/new' which should render me to the sign_up page. But it did not. Instead it throws an error.
![image](https://user-images.githubusercontent.com/38625956/82542390-b8ab4d80-9b73-11ea-88f6-b781cb138c8c.png)
So I think the line should be
```ruby
before_action :authenticate_user!, except: %i[index show]
```